shared/libraries/config_protocol/include/config_protocol/config_client_input_port_impl.h

```diff
@@ -53,21 +61,74 @@ inline ErrCode ConfigClientInputPortImpl::connect(ISignal* signal)
 {
     OPENDAQ_PARAM_NOT_NULL(signal);
 
-    return daqTry([this, &signal] {
-        const auto signalPtr = SignalPtr::Borrow(signal);
+    return daqTry(
+        [this, &signal]
+        {
+            if (!this->deserializationComplete)
+                return Super::connect(signal);
+
+            const auto signalPtr = SignalPtr::Borrow(signal);
+            {
+                std::scoped_lock lock(this->sync);
 
-        const auto configObject = signalPtr.asPtrOrNull<IConfigClientObject>(true);
-        if (!configObject.assigned())
-            throw InvalidParameterException("Not a remote signal");
+                if (isConnected(signalPtr))
+                    return OPENDAQ_IGNORED;
+            }
 
-        StringPtr signalRemoteGlobalId;
-        checkErrorInfo(configObject->getRemoteGlobalId(&signalRemoteGlobalId));
+            const auto configObject = signalPtr.asPtrOrNull<IConfigClientObject>(true);
+            if (!configObject.assigned())
+                throw InvalidParameterException("Not a remote signal");
```

Comment:
We are missing a check here that the signal is actually owned by the correct remote client device. We should at least note it down as a TODO so it's not missed.

Reply:
added TODO comment, but such signal would anyway be rejected by the device, assuming that global ID is globally unique.